### ISSUE TITLE
Devserver manifest is no longer observes visibility parameter passed in

### ISSUE THREAD

**Issue by achew22** [bazelbuild/rules_typescript#436]
# 🐞 bug report

### Affected Rule

`ts_devserver_macro`

The issue is caused by the rule setting the visibility of the internal [`ts_devserver`](https://github.com/bazelbuild/rules_typescript/blob/master/internal/devserver/ts_devserver.bzl#L244
) rule to private.

### Is this a regression?

Yes. The regression is from commit https://github.com/bazelbuild/rules_typescript/commit/dffa7a22161a8917742940706a8936651a1a1f00

Yes, the previous version in which this bug was not present was that you could access the .MF file generated by the macro.

### Description

The manifest should be visible to an external rule.

## 🔬 Minimal Reproduction

`BUILD`:

```
ts_library(
    name = "lib",
    srcs = ["hello.ts"],
)

ts_devserver(
    name = "hello",
   deps = [":lib"],
)

genrule(
    name = "fail",
    srcs = [":hello.MF"],
    cmd = "cat $(location :hello.MF)",
)
```

Leaving `hello.ts` empty should be sufficient.

## 🔥 Exception or Error

`target //path/to/hello.MF is not visible from //other/target`

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
    Ubuntu 18.04
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
Build label: 0.23.1
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Mon Mar 4 10:37:56 2019 (1551695876)
Build timestamp: 1551695876
Build timestamp as int: 1551695876
  </code>
</pre>

**Rules version (SHA):**

<pre>
  <code>
    "@bazel/typescript": "0.25.1",
  </code>
</pre>

**Anything else relevant?**

This change also renamed the output from `${TARGET}.MF` to `${TARGET}_launcher.MF` as a consequence of the wrapper `sh_binary` script.

**Comment by achew22** [comment]
This is a P1 since there is a simple workaround. You can add 

```
# Disable visibility because of rules typescript not exposing .MF files
build --nocheck_visibility
test --nocheck_visibility
run --nocheck_visibility
```

to your `.bazelrc`

**Comment by alexeagle** [comment]
Hey @achew22 could you include what is your use case for using the manifest file?
Another principled fix could be that we claim the manifest is a private API and change the docs to say so

**Comment by achew22** [comment]
Absolutely. I'm using the `.MF` to serve my resources in my devserver which is integrated with my server. This was the advised usage pattern for people who needed more than what devserver could provide. Is there a better way to do that now?
